src/scimlfunctions.jl

```diff
@@ -3890,3 +3890,28 @@ for S in [:ODEFunction
         end
     end
 end
+
+"""
+$(SIGNATURES)
+
+Converts a NonlinearFunction into a ODEFunction.
+"""
+function Base.convert(::Type{ODEFunction}, f::NonlinearFunction)
+    iip = isinplace(f)
+    _f = iip ? (du, u, p, t) -> (f.f(du, u, p); nothing) : (u, p, t) -> f.f(u, p)
+
+    ODEFunction{iip}(_f;
+                     mass_matrix = f.mass_matrix,
+                     analytic = f.analytic,
+                     tgrad = f.tgrad,
+                     jac = f.jac,
```

Comment:
this would need to add a `t` to it as well?

Reply:
oh yeah, good catch. I'll add it with some tests.